bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.digitalstrom.internal.lib.structure.devices.deviceparameters.impl;
+
+/**
+ * The {@link DSUID} represents the digitalSTROM-Device unique identifier.
+ *
+ * @author Alexander Friese - initial contributer
+ */
+public class DSUID {
+
+    private String dsuid;
+    private final String DEFAULT_DSUID = "3504175fe0000000000000000000000001";
+    private final String PRE = "3504175fe0000000";
+    private final String ALL = "ALL";
+
+    /**
+     * Creates a new {@link DSUID}.
+     *
+     * @param dsuid to create
+     */
+    public DSUID(String dsid) {
+        this.dsuid = dsid;
+        if (dsid != null && !dsid.trim().equals("")) {
```

Comment:
You could put the result of dsid.trim() in a variable as you called it several times.

Reply:
just copied from DSID class, now simplified a lot.